src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseCancellationTokenThrowIfCancellationRequested.cs

```diff
@@ -0,0 +1,203 @@
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+        internal readonly struct RequiredSymbols
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+        {
+            public static bool TryGetSymbols(Compilation compilation, out RequiredSymbols symbols)
+            {
+                symbols = default;
+                INamedTypeSymbol boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+                if (boolType is null)
+                    return false;
+
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingCancellationToken, out INamedTypeSymbol? cancellationTokenType))
+                    return false;
+
+                IMethodSymbol? throwIfCancellationRequestedMethod = cancellationTokenType.GetMembers(nameof(CancellationToken.ThrowIfCancellationRequested))
+                    .OfType<IMethodSymbol>()
+                    .GetFirstOrDefaultMemberWithParameterInfos();
+                IPropertySymbol? isCancellationRequestedProperty = cancellationTokenType.GetMembers(nameof(CancellationToken.IsCancellationRequested))
+                    .OfType<IPropertySymbol>()
+                    .FirstOrDefault();
+
+                if (throwIfCancellationRequestedMethod is null || isCancellationRequestedProperty is null)
+                    return false;
+
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemOperationCanceledException, out INamedTypeSymbol? operationCanceledExceptionType))
+                    return false;
+
+                IMethodSymbol? operationCanceledExceptionDefaultCtor = operationCanceledExceptionType.InstanceConstructors
+                    .GetFirstOrDefaultMemberWithParameterInfos();
+                IMethodSymbol? operationCanceledExceptionTokenCtor = operationCanceledExceptionType.InstanceConstructors
+                    .GetFirstOrDefaultMemberWithParameterInfos(ParameterInfo.GetParameterInfo(cancellationTokenType));
+
+                if (operationCanceledExceptionDefaultCtor is null || operationCanceledExceptionTokenCtor is null)
+                    return false;
+
+                symbols = new RequiredSymbols
+                {
+                    BoolType = boolType,
+                    CancellationTokenType = cancellationTokenType,
+                    OperationCanceledExceptionType = operationCanceledExceptionType,
+                    ThrowIfCancellationRequestedMethod = throwIfCancellationRequestedMethod,
+                    IsCancellationRequestedProperty = isCancellationRequestedProperty,
+                    OperationCanceledExceptionDefaultCtor = operationCanceledExceptionDefaultCtor,
+                    OperationCanceledExceptionTokenCtor = operationCanceledExceptionTokenCtor
+                };
+
+                return true;
+            }
+
+            public INamedTypeSymbol BoolType { get; init; }
```

Comment:
I  might be missing something, but this property doesn't seem to have used
```suggestion
```

Reply:
Actually it appears  `BoolType`, `CancellationTokenType`, `OperationCanceledExceptionType`, and `ThrowIfCancellationRequestedMethod` are all never used. I will remove them. 

To be clear, these are all symbols that we want to bail if we can't find them. However, the above 4 symbols aren't used during the actual analysis, so they can be removed as properties.